Kernel/Net/TCPSocket.cpp

```diff
@@ -360,18 +360,22 @@ bool TCPSocket::should_delay_next_ack() const
 
 NetworkOrdered<u16> TCPSocket::compute_tcp_checksum(IPv4Address const& source, IPv4Address const& destination, TCPPacket const& packet, u16 payload_size)
 {
-    struct [[gnu::packed]] PseudoHeader {
-        IPv4Address source;
-        IPv4Address destination;
-        u8 zero;
-        u8 protocol;
-        NetworkOrdered<u16> payload_size;
+    union PseudoHeader {
+        struct [[gnu::packed]] {
+            IPv4Address source;
+            IPv4Address destination;
+            u8 zero;
+            u8 protocol;
+            NetworkOrdered<u16> payload_size;
+        } header;
+        u16 raw[6];
     };
+    static_assert(sizeof(PseudoHeader) == 12);
 
-    PseudoHeader pseudo_header { source, destination, 0, (u8)IPv4Protocol::TCP, packet.header_size() + payload_size };
+    PseudoHeader pseudo_header { .header = { source, destination, 0, (u8)IPv4Protocol::TCP, packet.header_size() + payload_size } };
 
     u32 checksum = 0;
-    auto raw_pseudo_header = bit_cast<u16*>(&pseudo_header);
+    auto raw_pseudo_header = pseudo_header.raw;
```

Comment:
auto*

Reply:
Should we make the others `auto*` as well then? Or just the one I touched?

Reply:
As long as it's not too many, doing all of them should be fine.